lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/tool/producers/ShowPageProducer.java

```diff
@@ -5231,7 +5244,11 @@ private void createQuestionDialog(UIContainer tofill, SimplePage currentPage) {
 		UIBoundBoolean.make(form, "question-prerequisite", "#{simplePageBean.prerequisite}");
 		UIInput.make(form, "question-text-input", "#{simplePageBean.questionText}");
 		UIInput.make(form, "question-answer-full-shortanswer", "#{simplePageBean.questionAnswer}");
-		
+
+		UIOutput gradeBook = UIOutput.make(form, "gradeBookQuestionsDiv");
+		if(simplePageBean.getCurrentTool(simplePageBean.GRADEBOOK_TOOL_ID) == null) {
+			gradeBook.decorate(new UIStyleDecorator("noDisplay"));
+		}
```

Comment:
Instead of having one `if` for every component, just write one `if` at the end and `decorate` them all inside. This will reduce the code.

Reply:
The methods are different (different UI dialogs), I can only do it for createStudentContentDialog which has two of them.

Reply:
Ah, okay, I didn't notice that. Thanks!

Reply:
I removed the one if that can be shared